implementations/rust/ockam/ockam_api/src/kafka/key_exchange/controller.rs

```diff
@@ -127,20 +121,119 @@ impl KafkaKeyExchangeController for KafkaKeyExchangeControllerImpl {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub(crate) struct TopicEncryptionKeyHandler {
```

Comment:
Naming is hard but somehow I'm not sure if `TopicEncryptionKeyHandler` and `TopicEncryptionKey` are the best names for these (is one really a `Handler` or more like a `Handle`? Is the other one really a `Key` or some `Key` metadata, or some "rotation state"?). 
Maybe we could introduce the fact that one is for the consumer and the other one is for the producer? I'm just writing this in case you can think of alternatives.

Reply:
I agree, the naming is off, for now, `TopicEncryptionKey` became `TopicEncryptionKeyState` and `TopicEncryptionKeyHandler` became `TopicEncryptionKey`.
This code will probably change once the producer will not directly contact the consumer and the exchange happens via a different flow.